### PR TITLE
Modify get contained document to accept null id as input.

### PR DESCRIPTION
**Pull request by dibarbet**
Liveshare is broken for 16.1 P4 or greater clients because a null check was misplaced in a refactoring.

[This method](https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/Core/Def/Implementation/VisualStudioSupportsFeatureService.cs#L57) returns null because liveshare uses their own workspace type.  This causes [TryGetContainedDocument ](https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/Core/Def/Implementation/VisualStudioSupportsFeatureService.cs#L87)to throw an argument exception as the id is null.

The fix implementing here is to either modify all the call sites to skip this call when null or return null when passed in null to TryGetContainedDocument.  In all reference locations I could find, a null check is performed before the result of TryGetContainedDocument was used, so this seems safe (and it's already TryGet)

Tested in a liveshare session that a > 16.1p4 client can use preview code actions.

### CODE REVIEW

**Comment by dibarbet** [comment]
@daytonellwanger FYI

File: src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs [hunk 1]
```diff
@@ -62,6 +62,13 @@ internal sealed partial class ContainedDocument : ForegroundThreadAffinitizedObj
 
         public static ContainedDocument TryGetContainedDocument(DocumentId id)
         {
+            // Liveshare creates their own non-VS workspace which downloads files from the host.
+            // In such a case, a null id can be passed in, so check before attempting to retrieve the document.
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs]: This comment seems out of place to me. It feels like either:

1. The behavior makes sense for this method, and doesn't need further justification, or,
2. Really this is a terrible hack and maybe it shouldn't be here.

This feels like fixing a bug in int.Parse and then commenting that "it's needed for JSON parsing". This is a core bit of code -- it shouldn't be thinking about very specific clients or something seems off.

Comment (jasonmalinowski) [reply src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs]: (I'd go with 1 and say just remove the comment.)

Comment (dibarbet) [reply src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs]: Sure I can remove it - checking with @heejaechang are you fine without the comment?

File: src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs [hunk 2]
```diff
@@ -62,6 +62,13 @@ internal sealed partial class ContainedDocument : ForegroundThreadAffinitizedObj
 
         public static ContainedDocument TryGetContainedDocument(DocumentId id)
         {
+            // Liveshare creates their own non-VS workspace which downloads files from the host.
```
Comment (dibarbet) [inline src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs]:
```suggestion
```

File: src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs (hunk 1 above)
Comment (dibarbet) [inline src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs]:
```suggestion
```

**Pull request merged**